test/verify/check-realms

```diff
@@ -332,12 +421,19 @@ class TestIPA(TestRealms, CommonTests):
         super().setUp()
         self.admin_user = "admin"
         self.admin_password = "foobarfoo"
+        self.alice_password = 'WonderLand123'
         self.expected_server_software = "ipa"
         self.machines['services'].execute("/run-freeipa")
         # Wait for FreeIPA to come up and DNS to work as expected
         # https://bugzilla.redhat.com/show_bug.cgi?id=1071356#c11
         wait(lambda: self.machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
 
+        # wait until FreeIPA started up
+        self.machines['services'].execute("""docker exec -i freeipa sh -ec '
+            while ! echo %s | kinit -f %s; do sleep 5; done
+            while ! ipa user-find >/dev/null; do sleep 5; done'
+            """ % (self.admin_password, self.admin_user), timeout=300)
```

Comment:
I was about to browbeat you a little more for using `%` instead of `.format()` like the cool kids do, and decided to research the current 'pythonic' recommended best-practices here.  I found out about "f-strings", which let you write things like so:

`f"echo {self.admin_password} | kinit ..."`, which is more or less equivalent to the old `.format(**vars())` trick (which always seemed a bit dirty).  This new syntax feels really nice, though, as if it were actually intended as a proper language feature.

I feel like we could dramatically increase the readability of a lot of places in our tests with this...

And perhaps you won't hate it like you hate `str.format()` :rofl:

Reply:
Oh yes, f-strings are *the* answer, seriously. They were introduced into Python 3.6, and at least up until recently we couldn't rely on that as RHEL CI still had an older version. I can't say off-hand whether we can do this now, but I'd rather not do this just yet.